### PR TITLE
Add doc comments to OutputStream

### PR DESCRIPTION
**Pull request by pepone**
Add doc comments for OutputStream TypeScript definition.

### CODE REVIEW

File: js/src/Ice/Stream.d.ts [hunk 1]
```diff
@@ -135,64 +135,293 @@ declare module "ice" {
              */
             constructor(encoding?: EncodingVersion, format?: FormatType);
 
-            //
-            // This function allows this object to be reused, rather than reallocated.
-            //
-            reset(): void;
```
Comment (pepone) [inline js/src/Ice/Stream.d.ts:141]: This `reset` is never used in JS.

File: js/src/Ice/Stream.d.ts [hunk 2]
```diff
@@ -135,64 +135,293 @@ declare module "ice" {
...
+             * @returns The Uint8Array containing the encoded request or reply
+             */
             finished(): Uint8Array;
+
+            /**
+             * Swaps the contents of one stream with another.
+             * @param other The other stream.
+             */
             swap(other: OutputStream): void;
-            resetEncapsulation(): void;
```
Comment (pepone) [inline js/src/Ice/Stream.d.ts:145]: `resetEncapsulation` is not part of the public API.

File: js/src/Ice/Stream.d.ts [hunk 3]
```diff
@@ -135,64 +135,293 @@ declare module "ice" {
...
+             */
             writeValue(v: Ice.Value): void;
+
+            /**
+             * Writes a user exception to the stream.
+             *
+             * @param v The user exception to write.</param>
+             */
             writeException(e: UserException): void;
-            writeUserException(e: UserException): void;
```
Comment (pepone) [inline js/src/Ice/Stream.d.ts:184]: Removed `writeUserException`, deprecated in 3.7.

**Review by externl — APPROVED** [review]

File: js/src/Ice/Stream.d.ts [hunk 4]
```diff
@@ -135,64 +135,293 @@ declare module "ice" {
...
+             *
+             * @param n The number of bytes to accommodate in the stream.
+             */
             expand(n: number): void;
 
-            //
-            // Sets the encoding format for class and exception instances.
-            //
+            /**
+             * Gets or sets the encoding format for class and exception instances.
```
Comment (bernardnormier) [inline js/src/Ice/Stream.d.ts:408]: Are you sure this is needed? _format is read-only and private in other language mappings.

**Review by bernardnormier — APPROVED** [review]

File: js/src/Ice/Stream.d.ts (hunk 4 above)
Comment (pepone) [reply js/src/Ice/Stream.d.ts]: Removed.

**Pull request merged**